lib/dpif-netdev.c

```diff
@@ -4587,6 +4594,26 @@ rxq_scheduling(struct dp_netdev *dp, bool pinned) OVS_REQUIRES(dp->port_mutex)
         }
     }
 
+    /* Set pmd core pin of the rxq soft affinity port . */
+    HMAP_FOR_EACH (port, node, &dp->ports) {
+        if (!netdev_is_pmd(port->netdev)) {
+            continue;
+        }
+
+        for (int qid = 0; qid < port->n_rxq; qid++) {
+            struct dp_netdev_rxq *q = &port->rxqs[qid];
+
+            if (pinned && q->core_id != OVS_CORE_UNSPEC) {
+                struct dp_netdev_pmd_thread *pmd;
+
+                pmd = dp_netdev_get_pmd(dp, q->core_id);
+                if (q->port->rxq_soft_affinity){
+                    q->pmd = pmd;
```

Comment:
This part of the code will break the load balancing that happens in this function few lines below, because it doesn't know that some pmd threads already has some queues assigned.